### DISCUSSION TITLE
Unable to install 2.0 beta by following documentation

### DISCUSSION THREAD

**Discussion by memark** [sqlalchemy/sqlalchemy#7623] [Usage Questions]
### Describe the bug

Documentation for 2.0 states:

> In order to install the latest prerelease version, *such as 2.0.0b1*, pip requires that the --pre flag be used:
> 
> pip install --pre SQLAlchemy

The suggested `--pre` flag has no effect. 1.4 is installed anyway.

### To Reproduce

```python
❯ docker run --rm -it python bash
root@e8665c311d75:/# pip install --pre SQLAlchemy
Collecting sqlalchemy
  Downloading SQLAlchemy-1.4.31-cp310-cp310-manylinux_2_17_aarch64.manylinux2014_aarch64.whl (1.6 MB)
     |████████████████████████████████| 1.6 MB 1.8 MB/s
Collecting greenlet!=0.4.17
  Downloading greenlet-2.0.0a1-cp310-cp310-manylinux_2_17_aarch64.manylinux2014_aarch64.whl (516 kB)
     |████████████████████████████████| 516 kB 28.9 MB/s
Installing collected packages: greenlet, sqlalchemy
Successfully installed greenlet-2.0.0a1 sqlalchemy-1.4.31
```

### Error

```
Successfully installed greenlet-2.0.0a1 sqlalchemy-1.4.31
``` 

### Versions

- OS: MacOS 12.1
- Python: 3.10.1
- SQLAlchemy: 2.0
- Database: n/a
- DBAPI (eg: psycopg, cx_oracle, mysqlclient): n/a

**Comment by CaselIT** [comment]
Hi,

This was litterally asked 1 hour ago https://github.com/sqlalchemy/sqlalchemy/discussions/7619

There is no current beta on pypy. You need to install using the git url

**Comment by CaselIT** [comment]
I guess we can improve the docs here, since it does not seem clear

**Comment by memark** [comment]
Hi @CaselIT! Thank you for your rapid response.

I was not aware of the discussions tab having similar questions. I searched both open and closed issues before creating this one.

OK, so it's not available on pypi. In that case it's misleading (for me) to have an example on how to install it with pip! I would question the usefulness of those sentences when we already know that command won't work. (I refer to the explanation of how to install 2.0.0b1.)

Could we not make it available on pypi? Pypi has functionality to handle prerelease versions AFAIK. I think that would make it much easier to get feedback on the 2.0 development.

Cloning the git repo is not technically advanced, but it adds to much complexity both locally and in development pipelines. Being able to install a beta with --pre would be the dealbraker here for me and my team.

If 2.0 was just some far away internal development version I wouldn't mind. But the 1.4 docs mention 2.0 a lot, so I thought I'd get on the wagon early and help out testing the new version. I would suggest choosing a clear strategy here, either talking a lot about 2.0 and making a prerelease easily available, or toning down its profile and keeping it hidden for the time being.

Just my two cents. Keep up the good work!

**Comment by CaselIT** [comment]
> OK, so it's not available on pypi. In that case it's misleading (for me) to have an example on how to install it with pip! I would question the usefulness of those sentences when we already know that command won't work. (I refer to the explanation of how to install 2.0.0b1.)

these are general docs, taken from the 1.4 series. The documentation for 2.0 says "in development", not "in beta".



> Could we not make it available on pypi? Pypi has functionality to handle prerelease versions AFAIK. I think that would make it much easier to get feedback on the 2.0 development.

there is still nothing ready to release on pypi. once there is we will release it



> Cloning the git repo is not technically advanced, but it adds to much complexity both locally and in development pipelines. Being able to install a beta with --pre would be the dealbraker here for me and my team.

again there is no beta to install, so if you want to install the development version git is the right tool for it

> If 2.0 was just some far away internal development version I wouldn't mind. But the 1.4 docs mention 2.0 a lot, so I thought I'd get on the wagon early and help out testing the new version. I would suggest choosing a clear strategy here, either talking a lot about 2.0 and making a prerelease easily available, or toning down its profile and keeping it hidden for the time being.

I believe the plan is outlined quite well in the documentation already, there is a long page detailing it: https://docs.sqlalchemy.org/en/14/changelog/migration_20.html

**Comment by memark** [comment]
Most people would interpret "Version 2.0 (development)" on the website as if the 2.0 version is in development. Not that the _documentation itself_ is in development and can't be relied on even when it comes to how to install the software.

>I believe the plan is outlined quite well in the documentation already, there is a long page detailing it: https://docs.sqlalchemy.org/en/14/changelog/migration_20.html

The migration docs are great from a code perspective! But they mention nothing about installation, so I don't see them applicable here.

>again there is no beta to install, so if you want to install the development version git is the right tool for it

Since you tagged this issue with "bug" and "documentation" I assume you actually want to improve the docs. The crux of the matter is that it's confusing to have the docs specifically mention how to install the 2.0 BETA via pip when no beta is even available. Remove that part, add "not available on Pypi", point to "git clone" as the only installation method for 2.0, and you will stop getting questions like this.

**Comment by CaselIT** [comment]
My plan was to add that once a version is in beta the pre-release can be downloaded using `--pre`, while when in develop it should be installed using the git url

**Comment by zzzeek** [comment]
the upper right corner of the docs looks like this:

      Release: 2.0.0b1 in development | Release Date: not released yet

we will update the "--pre" docs right now with a dragon warning.   the new bug then will be we will haev no way to remember how to take it off unless we automate it, but i dont have time to automate that right now.
